hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java

```diff
@@ -80,9 +81,15 @@ private OzoneClient getClient(OzoneConfiguration config) throws IOException {
       UserGroupInformation remoteUser =
```

Comment:
This will throw NPE when awsAccessId is null, do we need to move validateAccessId before remoteUser creation.

Reply:
Sure. Updated.